vic/drivers/shared_image/src/vic_restore.c

```diff
@@ -0,0 +1,1009 @@
+        for (i = 0; i < local_domain.ncells_active; i++) {
+            all_vars[i].lake_var.coldcontent = dvar[i];
+        }
+
+        // lake ice snow surface water: lake_var.surf_water
+        get_scatter_nc_field_double(filenames.init_state,
+                                    "Lake_ice_snow_surf_water",
+                                    d2start, d2count, dvar);
+        for (i = 0; i < local_domain.ncells_active; i++) {
+            all_vars[i].lake_var.surf_water = dvar[i];
+        }
+
+        // lake ice snow pack water: lake_var.pack_water
+        get_scatter_nc_field_double(filenames.init_state,
+                                    "Lake_ice_snow_pack_water",
+                                    d2start, d2count, dvar);
+        for (i = 0; i < local_domain.ncells_active; i++) {
+            all_vars[i].lake_var.pack_water = dvar[i];
+        }
+
+        // lake ice snow albedo: lake_var.SAlbedo
+        get_scatter_nc_field_double(filenames.init_state,
+                                    "Lake_ice_snow_albedo",
+                                    d2start, d2count, dvar);
+        for (i = 0; i < local_domain.ncells_active; i++) {
+            all_vars[i].lake_var.SAlbedo = dvar[i];
+        }
+
+        // lake ice snow depth: lake_var.sdepth
+        get_scatter_nc_field_double(filenames.init_state,
+                                    "Lake_ice_snow_depth",
+                                    d2start, d2count, dvar);
+        for (i = 0; i < local_domain.ncells_active; i++) {
+            all_vars[i].lake_var.sdepth = dvar[i];
+        }
+    }
+
+    free(cvar);
+    free(ivar);
+    free(dvar);
+    free(fvar);
+}
+
+void
+check_init_state_file(void)
+{
+    extern filenames_struct filenames;
+    extern domain_struct    global_domain;
+    extern option_struct    options;
+    extern soil_con_struct *soil_con;
+
+    int                     status;
+    size_t                  dimlen;
+    size_t                  i;
+    size_t                  d1count[1];
+    size_t                  d1start[1];
+    size_t                  d2count[2];
+    size_t                  d2start[2];
+    int                     lon_var_id;
+    int                     lat_var_id;
+    double                 *dvar;
+    nc_file_struct          nc;
+
+    // open the netcdf file
+    status = nc_open(filenames.init_state, NC_SHARE, &(nc.nc_id));
+    if (status != NC_NOERR) {
+        log_err("Error opening %s", filenames.init_state);
+    }
+
+    // read and validate dimension lengths
+    dimlen = get_nc_dimension(filenames.init_state, "ni");
+    if (dimlen != global_domain.n_nx) {
+        log_err("Number of grid columns in state file does not "
+                "match parameter file");
+    }
+    dimlen = get_nc_dimension(filenames.init_state, "nj");
+    if (dimlen != global_domain.n_ny) {
+        log_err("Number of grid rows in state file does not "
+                "match parameter file");
+    }
```

Comment:
OK to the first sentence.  Although we'll have to document somewhere that VIC won't be able to read a state file whose dimension names don't match those of the forcing and parameter files.  Obviously in most cases VIC will create the state file to match, but if someone creates a state file outside of VIC or tonic, they'll need to know that their dimension names must match those of the forcings and parameters that they're using.  Am I understanding this correctly?

What if someone uses forcings from one source (say, NARR) and parameters from some other source (say, Livneh et al 2013).  Will they need to modify the dimension names to all be exactly the same?  (I realize that in mixing/matching like that, probably other modifications would also need to be made to the files anyway).

I wonder if it wouldn't be better to have separate sets of dimension names, ids, etc for forcing, parameter, and state files (and any other input files).  These would all need to be specified by the user in the global parameter file.  Or some logic would have to be inserted into VIC to automatically detect these, but that seems to be too much.